File: Codes/Hash/Non-Cryptographic/DJBHash/DJBHash.cs
/*
	Hash function by Daniel J. Bernstein
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static uint DJBHash (string key)
{
    uint state = 5381;
	uint i = 0;

	for (i = 0; i < key.Length; i++)
	{
		state = ((state << 5) + state) + ((byte)key[(int)i]);
	}

	return state;
}

File: Codes/Hash/Non-Cryptographic/DJBHash/DJBHash.vb
'  Hash function by Daniel J. Bernstein
'  Archive of Reversing.ID
'  Non-Cryptographic Hash
'

' Fungsi ini merupakan method dari sebuah class
Public Shared Function DJBHash (key As String) As UInteger
    Dim state As ULong = 5381
	Dim i As UInteger = 0

	For i = 0 To key.Length - 1
		state = (((state << 5) + state) + CByte(AscW(key(CInt(i)))) And UInteger.MaxValue)
	Next

	Return state
End Function